src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -172,21 +172,21 @@
                            :on-selection-change      (partial on-selection-change timeout-id last-text-change mentionable-users)
                            :on-change                (partial on-change last-text-change timeout-id mentionable-users refs chat-id sending-image)
                            :on-text-input            (partial on-text-input mentionable-users chat-id)}
-        children          (fn []
-                            (if mentions-enabled
-                              (for [[index [type text]] (map-indexed
-                                                         (fn [idx item]
-                                                           [idx item])
-                                                         (<sub [:chat/input-with-mentions]))]
-                                ^{:key (str index "_" type "_" text)}
-                                [rn/text (when (= type :mention) {:style {:color colors/primary-50}})
-                                 text])
-                              (get @input-texts chat-id)))]
+        children          (reagent/as-element
+                           (if mentions-enabled
+                             (for [[index [type text]] (map-indexed
```

Comment:
I know you don't want to change the original implementation much, but the `for` macro surrounding the `map-indexed` seems unnecessary, i.e. we're looping twice.

```clojure
(map-indexed (fn [index [type text]]
               ^{:key (str index "_" type "_" text)}
               [rn/text (when (= type :mention) {:style {:color colors/primary-50}})
                text])
             (<sub [:chat/input-with-mentions]))
```

Reply:
Wow, I don't how I didn't notice this
Thanks Icaro!